bundles/org.openhab.binding.irobot/src/main/java/org/openhab/binding/irobot/internal/RawMQTT.java

```diff
@@ -0,0 +1,187 @@
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * A "raw MQTT" client for sending custom "get password" request.
+ * Seems pretty much reinventing a bicycle, but it looks like HiveMq
+ * doesn't provide for sending and receiving custom packets.
+ *
+ * @author Pavel Fedin <pavel_fedin@mail.ru> - Initial contribution
+ *
+ */
+@NonNullByDefault
+public class RawMQTT {
+    public static final int ROOMBA_MQTT_PORT = 8883;
+
+    private Socket socket;
+
+    public static class Packet {
+        public byte message;
+        public byte[] payload;
+
+        Packet(byte msg, byte[] data) {
+            message = msg;
+            payload = data;
+        }
+
+        public boolean isValidPasswdPacket() {
+            return message == PasswdPacket.MESSAGE && payload.length >= PasswdPacket.HEADER_SIZE;
+        }
+    };
+
+    public static class PasswdPacket extends Packet {
+        static final byte MESSAGE = (byte) 0xF0; // MQTT Reserved
+        static final int MAGIC = 0x293bccef;
+        static final byte HEADER_SIZE = 5;
+        private ByteBuffer buffer;
+
+        public PasswdPacket(Packet raw) {
+            super(raw.message, raw.payload);
+            buffer = ByteBuffer.wrap(payload).order(ByteOrder.LITTLE_ENDIAN);
+        }
+
+        public int getMagic() {
+            return buffer.getInt(0);
+        }
+
+        public byte getStatus() {
+            return buffer.get(4);
+        }
+
+        public @Nullable String getPassword() {
+            if (getStatus() != 0) {
+                return null;
+            }
+
+            int length = payload.length - HEADER_SIZE;
+            byte[] passwd = new byte[length];
+
+            buffer.position(HEADER_SIZE);
+            buffer.get(passwd);
+
+            return new String(passwd);
+        }
+    }
+
+    // Roomba MQTT is using SSL with custom root CA certificate.
+    private static class MQTTTrustManager implements X509TrustManager {
+        @Override
+        public X509Certificate @Nullable [] getAcceptedIssuers() {
+            return null;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate @Nullable [] arg0, @Nullable String arg1)
+                throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate @Nullable [] certs, @Nullable String authMethod)
+                throws CertificateException {
+            /*
+             * TODO: Retrieve Roomba CA certificate and implement proper verification
+             * logger.debug("Auth method: " + authMethod);
+             * for (X509Certificate cert : certs) {
+             * logger.debug("Cert: " + cert.toString());
+             * }
+             */
```

Comment:
Can you do this or remove the comment?

Reply:
Removed. It would take significant time to do it. Perhaps need to reverse engineer smartphone's app built-in check